EdvClient.js

```diff
@@ -347,38 +347,47 @@ export class EdvClient {
    * Deletes a document from the EDV.
    *
    * @param {Object} options - The options to use.
-   * @param {string} options.id the ID of the document to delete.
+   * @param {string} options.doc the document to delete.
+   * @param {Object} [options.recipients=[]] a set of JWE recipients to encrypt
+   *   the document for; if present, recipients will be added to any existing
+   *   recipients; to remove existing recipients, modify the
+   *   `encryptedDoc.jwe.recipients` field.
    * @param {string} [options.capability=undefined] - The OCAP-LD authorization
    *   capability to use to authorize the invocation of this operation.
    * @param {Object} options.invocationSigner - An API with an
    *   `id` property and a `sign` function for signing a capability invocation.
+   * @param {function} [keyResolver=this.keyResolver] a function that returns
+   *   a Promise that resolves a key ID to a DH public key.
+   * @param {Object} [keyAgreementKey=null] a default KeyAgreementKey API for
+   *   deriving shared KEKs for wrapping content encryption keys.
    *
-   * @return {Promise<Boolean>} resolves to `true` if the document was deleted
-   *   and `false` if it did not exist.
+   * @return {Promise<Boolean>} resolves to `true` when the document was
+   *  deleted.
    */
-  async delete({id, capability, invocationSigner}) {
-    _assertString(id, '"id" must be a string.');
+  async delete({
+    doc, recipients = [], capability, invocationSigner,
+    keyResolver = this.keyResolver, keyAgreementKey = this.keyAgreementKey}) {
+    _assertDocument(doc);
+    _assertString(doc.id, '"id" must be a string.');
     _assertInvocationSigner(invocationSigner);
 
-    const url = this._getDocUrl(id, capability);
-    if(!capability) {
-      capability = this._getRootDocCapability(id);
+    doc.content = {};
+    if(!doc.meta) {
+      doc.meta = {};
     }
```

Comment:
This will ensure `meta` is shallow copied (in case it's already there) before setting `deleted` to `true`.

```suggestion
    if(!doc.meta) {
      doc.meta = {};
    } else {
      doc.meta = {...doc.meta};
    }
```